### PR TITLE
ARROW-13239: [Python] [Doc] Expose signatures in pyx modules

### PR DESCRIPTION
**Pull request by amol-**

### CODE REVIEW

**Comment by kszucs** [comment]
Shell we include other directives too, similarly to [lib.pyx](https://github.com/apache/arrow/blob/master/python/pyarrow/lib.pyx#L18-L21)? Perhaps there is a way to set those globally from `setup.py`, though I'm not sure.

**Comment by amol-** [comment]
> Shell we include other directives too, similarly to [lib.pyx](https://github.com/apache/arrow/blob/master/python/pyarrow/lib.pyx#L18-L21)? Perhaps there is a way to set those globally from `setup.py`, though I'm not sure.

That's an interesting idea, there might be a way to pass it as a compiler option, in general providing them to `Extension` is the way to go, but the fact that we use CMake to cythonize files is something I'm not much familiar with so I'd have to investigate a bit how to pass compiler options through CMake

**Comment by amol-** [comment]
@kszucs I pursued your idea and now the signatures are generated for all `.pyx` files independently from the directives applied at the top of the file. That should address the issue and prevent future problems from occurring again.

File: cpp/cmake_modules/UseCython.cmake [hunk 1]
```diff
@@ -145,6 +145,8 @@ function(compile_pyx
             ${no_docstrings_arg}
             ${cython_debug_arg}
             ${CYTHON_FLAGS}
+            # Necessary for autodoc of function arguments
+            --directive embedsignature=True
```
Comment (kszucs) [inline cpp/cmake_modules/UseCython.cmake:149]: Nice! Can we remove the corresponding directives from the cython files?

Comment (amol-) [reply cpp/cmake_modules/UseCython.cmake:149]: done, remove the directive from the files that had it

**Comment by amol-** [comment]
@jorisvandenbossche @kszucs do you mind double checking this one to confirm it's the way to go? I think it would be great to fix the api reference for 5.0.0

**Review by jorisvandenbossche — APPROVED** [review]
Looks good! (I assume you checked the doc build with this?)

**Pull request closed without merging**